lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -114,35 +114,47 @@ class ConvertAtenBinaryOp : public OpConversionPattern<AtenOpT> {
 // FIXME: This will eventually go into a Tosa*Utils file.
 LogicalResult torchScalarToTosaTensor(ConversionPatternRewriter &rewriter,
                                       Operation *op, Value torchScalarValue,
-                                      Value &tosaTensor, Type dtype) {
-  if (dtype.isa<mlir::FloatType>()) {
-    double scalarValue;
-
-    if (!matchPattern(torchScalarValue, m_TorchConstantFloat(&scalarValue)))
-      return failure();
+                                      Value &tosaTensor, Type dtype,
+                                      llvm::ArrayRef<int64_t> dshape) {
+  // Retrieve a const float or int value but create the out Tensor with dtype.
+  double doubleValue;
+  auto is_float =
```

Comment:
nit: camelBack is being used in this file so `isFloat` would be more consistent.